vpp-manager/vpp_runner.go

```diff
@@ -697,24 +699,38 @@ func (v *VppRunner) pingCalicoVpp() error {
 
 // Returns VPP exit code
 func (v *VppRunner) runVpp() (err error) {
-	// From this point it is very important that every exit path calls restoreConfiguration after vpp exits
-	vppCmd := exec.Command(config.VppPath, "-c", config.VppConfigFile)
-	vppCmd.SysProcAttr = &syscall.SysProcAttr{
-		// Run VPP in an isolated network namespace, used to park the interface in
-		// af_packet or af_xdp mode
-		Cloneflags: syscall.CLONE_NEWNET,
+	var vppPid int = -1
+
+	if ns.IsNSorErr(utils.GetnetnsPath(config.VppNetnsName)) != nil {
+		_, err = utils.NewNS(config.VppNetnsName)
+		if err != nil {
+			return errors.Wrap(err, "Could not add VPP netns")
+		}
 	}
-	vppCmd.Stdout = os.Stdout
-	vppCmd.Stderr = os.Stderr
-	err = vppCmd.Start()
 
+	/**
+	 * From this point it is very important that every exit
+	 * path calls restoreConfiguration after vpp exits */
 	defer v.restoreConfiguration()
```

Comment:
Is this necessary before we start vpp now?

Reply:
I didn't see any drawbacks to running it even if VPP startup fails,
This allows to restore connectivity in the case VPP manages to grab the
uplink (e.g. dpdk), and still fail to start afterwards (e.g. oom in another plugin init)
Although this can arguably sound farfetched 😄